Packs/CheckpointFirewall/Playbooks/Checkpoint_-_Block_IP_-_Custom_Block_Rule.yml

```diff
@@ -725,183 +789,321 @@ tasks:
     view: |-
       {
         "position": {
-          "x": -1460,
-          "y": 2390
+          "x": -60,
+          "y": 780
         }
       }
     note: false
     timertriggers: []
     ignoreworker: false
-  "24":
-    id: "24"
-    taskid: 62b3be87-8644-4eda-8609-ae6504375e7c
-    type: regular
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "30":
+    id: "30"
+    taskid: c36c7553-cdc1-4850-8252-64ea6025fa6d
+    type: playbook
     task:
-      id: 62b3be87-8644-4eda-8609-ae6504375e7c
+      id: c36c7553-cdc1-4850-8252-64ea6025fa6d
       version: -1
-      name: Fix error and publish manually
-      description: User manually add resolve .
-      type: regular
+      name: Checkpoint - Block IP - Append Group
+      description: |-
+        This playbook will use an object group in order to block IPs.
+        The playbook receives malicious IP addresses as inputs, checks if the object group exists (and if not will create one), and will append the related IPs in that object.
+      playbookName: Checkpoint - Block IP - Append Group
+      type: playbook
       iscommand: false
       brand: ""
     nexttasks:
       '#none#':
-      - "2"
-    separatecontext: false
+      - "7"
+    scriptarguments:
+      IP:
+        simple: ${inputs.IP}
+      block_IP_error_handling:
+        simple: Stop
+      checkpoint_error_handling:
+        simple: Stop
+      group_name:
+        simple: XSOAR-${incident.id}
+      install_policy:
+        simple: "False"
+      policy_package:
+        simple: Standard
+    separatecontext: true
+    loop:
+      iscommand: false
+      exitCondition: ""
+      wait: 1
+      max: 100
     view: |-
       {
         "position": {
-          "x": -1460,
-          "y": 1540
+          "x": 460,
+          "y": 780
         }
       }
     note: false
     timertriggers: []
     ignoreworker: false
-  "25":
-    id: "25"
-    taskid: b2bfee2e-854f-4f48-8e56-dfe26e9ceea7
+    skipunavailable: true
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "31":
+    id: "31"
+    taskid: dd59fb80-7da3-44f6-8007-dc28d3e09809
     type: regular
     task:
-      id: b2bfee2e-854f-4f48-8e56-dfe26e9ceea7
+      id: dd59fb80-7da3-44f6-8007-dc28d3e09809
       version: -1
-      name: Fix error and publish manually
-      description: User manually add resolve .
+      name: Check XSOAR version
+      description: Return the Demisto server version.
+      scriptName: DemistoVersion
       type: regular
       iscommand: false
       brand: ""
     nexttasks:
       '#none#':
-      - "2"
+      - "32"
+    separatecontext: false
+    view: |-
+      {
+        "position": {
+          "x": 230,
+          "y": -500
+        }
+      }
+    note: false
+    timertriggers: []
+    ignoreworker: false
+    skipunavailable: false
+    quietmode: 0
+    isoversize: false
+    isautoswitchedtoquietmode: false
+  "32":
```

Comment:
I dont follow the solution here
I thought your suggestion was to create an additional playbook with from version 6.0 that will use the additional subplaybook and keep the previous 5.0 version without it

Reply:
And that's what it is - for 6.0 and above, the route will be a new playbook "Checkpoint - Block IP - Append group" on task # 30. The other route (default) will proceed with the old logic (which is not supporting an array at all).

Reply:
![image](https://user-images.githubusercontent.com/88268646/147733026-899b2384-f1c5-408a-bc57-8e2761b62c92.png)

Reply:
Changed it to support array - no need for this workaround.